api/src/test/java/org/openmrs/module/datafilter/impl/EncTypePrivilegeBasedFilterTest.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.datafilter.impl;
+
+import org.hibernate.SessionFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.EncounterType;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.datafilter.TestConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import java.util.List;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class EncTypePrivilegeBasedFilterTest extends BaseEncTypeViewPrivilegeBasedFilterTest {
+	
+	@Autowired
+	private SessionFactory sessionFactory;
+	
+	@Before
+	public void before() {
+		executeDataSet(TestConstants.ROOT_PACKAGE_DIR + "privilegedEncounters.xml");
+	}
+	
+	@Test
+	public void getEncounterType_shouldIncludeEncountersThatRequireAPrivilegeAndTheUserHasIt() {
+		reloginAs("dyorke", "test");
+		assertFalse(Context.getAuthenticatedUser().hasPrivilege(PRIV_MANAGE_CHEMO_PATIENTS));
+		
+		int expCount = 3;
+		List ecounterTypelist = sessionFactory.getCurrentSession().createCriteria(EncounterType.class).list();
```

Comment:
These lines show that you are not running your tests with a built in memory database, you could actually need to use data sets for the tests you are writing. @dkayiwa  what do you think?

Reply:
@HerbertYiga how do you tell that he is not a built in memory database?

Reply:
> @HerbertYiga how do you tell that he is not a built in memory database?

@dkayiwa on seeing the use of sessionFactory ,since this contains parameters for the external data base that is always connected too

Reply:
@HerbertYiga the parameters for session factory can be for any database whether in memory or not.